core/src/main/java/org/pentaho/di/connections/ConnectionDetails.java

```diff
@@ -67,10 +67,6 @@ default void setDescription( String description ) {
 
   void setSpace( VariableSpace space );
 
-  default boolean hasBuckets() {
```

Comment:
Taking the chance to move this to `VFSConnectionDetails`. Buckets only apply to VFS connections.

Reply:
@abryant-hv raised general concern about breaking compatibility: https://github.com/pentaho/pentaho-kettle/pull/9334#pullrequestreview-2071681357.

This one is a breaking change. The `hasBuckets` method has always been misplaced in the `ConnectionDetails` interface, and not in `VFSConnectionDetails` interface, given it is a VFS-only feature of connections.

It's unlikely this breaks any real third-party implementations though. The method was a `default` method. Also, it's unlikely that it was being overridden by any connection class which doesn't implement `VFSConnectionDetails`.

Concluding, only classes implementing `ConnectionDetails` and not `VFSConnectionDetails` and which override the method and and explicitly annotate it with `@Override` will break.

Reply:
One alternative would be to proceed as suggested by you in https://github.com/pentaho/pentaho-kettle/pull/9334/files#r1610245712, to deprecate this one and re-declare/override it in `VFSConnectionDetails`.